bindings/go/rsa/signing/handler/internal/dn/dn.go

```diff
@@ -0,0 +1,117 @@
+// Package dn provides utilities for parsing and matching X.509 Distinguished Names (DNs).
+package dn
+
+import (
+	"crypto/x509/pkix"
+	"fmt"
+	"slices"
+	"strings"
+)
+
+// attributes maps DN attribute keys to functions that update a pkix.Name.
+var attributes = map[string]func(*pkix.Name, string){
+	"C":          func(n *pkix.Name, v string) { n.Country = append(n.Country, v) },
+	"O":          func(n *pkix.Name, v string) { n.Organization = append(n.Organization, v) },
+	"OU":         func(n *pkix.Name, v string) { n.OrganizationalUnit = append(n.OrganizationalUnit, v) },
+	"L":          func(n *pkix.Name, v string) { n.Locality = append(n.Locality, v) },
+	"ST":         func(n *pkix.Name, v string) { n.Province = append(n.Province, v) },
+	"STREET":     func(n *pkix.Name, v string) { n.StreetAddress = append(n.StreetAddress, v) },
+	"POSTALCODE": func(n *pkix.Name, v string) { n.PostalCode = append(n.PostalCode, v) },
+	"SN":         func(n *pkix.Name, v string) { n.SerialNumber = v },
+	"CN":         func(n *pkix.Name, v string) { n.CommonName = v },
+}
+
+// dnSplitRunes defines characters used to split DN components.
+var dnSplitRunes = []rune{'/', ';', ',', '+'}
+
+// Parse converts a string representation of a distinguished name into a pkix.Name.
```

Comment:
Nit: Also add the comment about the fallback behaviour at the end of this function.